rust/gui-client/src-tauri/src/client/gui.rs

```diff
@@ -638,13 +638,16 @@ impl Controller {
                 bail!("Impossible error: `Quit` should be handled before this")
             }
             Req::TunnelReady => {
+                let was_already_ready = self.tunnel_ready;
                 self.tunnel_ready = true;
                 self.refresh_system_tray_menu()?;
 
-                os::show_notification(
-                    "Firezone connected",
-                    "You are now signed in and able to access resources.",
-                )?;
+                if !was_already_ready {
```

Comment:
Maybe we can move this check before mutating `tunnel_ready` so we don't need the intermediate variable?

I don't feel to strongly about this

Reply:
Yeah, not sure why I had that variable there. Fixed it